src/coinbase/sponsored_send.ts

```diff
@@ -0,0 +1,126 @@
+import { ethers } from "ethers";
+import { SponsoredSend as SponsoredSendModel } from "../client/api";
+import { SponsoredSendStatus } from "./types";
+
+/**
+ * A representation of an onchain Sponsored Send.
+ */
+export class SponsoredSend {
+  private model: SponsoredSendModel;
+
+  /**
+   * Sponsored Sends should be constructed via higher level abstractions like Transfer.
+   *
+   * @class
+   * @param model - The underlying Sponsored Send object.
+   */
+  constructor(model: SponsoredSendModel) {
+    if (!model) {
+      throw new Error("Invalid model type");
+    }
+    this.model = model;
+  }
+
+  /**
+   * Returns the Keccak256 hash of the typed data. This payload must be signed
+   * by the sender to be used as an approval in the EIP-3009 transaction.
+   *
+   * @returns The Keccak256 hash of the typed data.
+   */
+  getTypedDataHash(): string {
+    return this.model.typed_data_hash;
+  }
+
+  /**
+   * Returns the signature of the typed data.
+   *
+   * @returns The Keccak256 hash of the typed data.
+   */
+  getSignature(): string | undefined {
+    return this.model.signature;
+  }
+
+  /**
+   * Signs the Sponsored Send with the provided key and returns the hex signature.
+   *
+   * @param key - The key to sign the Sponsored Send with
+   * @returns The hex-encoded signature
+   */
+  async sign(key: ethers.Wallet) {
+    ethers.toBeArray;
+    const signature = key.signingKey.sign(ethers.getBytes(this.getTypedDataHash())).serialized;
+    this.model.signature = signature;
+    // Removes the '0x' prefix as required by the API.
+    return signature.slice(2);
+  }
+
+  /**
+   * Returns whether the Sponsored Send has been signed.
+   *
+   * @returns if the Sponsored Send has been signed.
+   */
+  isSigned(): boolean {
+    return this.getSignature() ? true : false;
+  }
+
+  /**
+   * Returns the Status of the Sponsored Send.
+   *
+   * @returns the Status of the Sponsored Send
+   */
+  getStatus(): SponsoredSendStatus | undefined {
+    switch (this.model.status) {
```

Comment:
I don't see any differences in the actual switch, is this appreciably different than just returning `return this.model.status`?

Reply:
This is mainly used to type cast to the `SponsoredSendStatus` enum vs using const strings. The enum type is easier to use.